### DISCUSSION TITLE
Why are the tools to evaluate the models functions in a second package?

### DISCUSSION THREAD

**Discussion by DaveParr** [Nixtla/statsforecast#661] [Ideas]
I've been trying to work out for a while why statsforecast seems so feature complete, but that the methods to evalute the model performance are in a second, separate repo? 

`datasetsforecast/losses.py` seems to be the place that error metrics (mase, rmse, mape etc) are implemented in the nixtlaverse, but it also seems they are undertooled, in that they are stand alone functions that are not tightly coupled to the nixtlaverse data structures. 

Would there be interest/scope to implement something like:

``` python
from statsforecast import StatsForecast
from statsforecast.models import ADIDA, ARCH

sf = StatsForecast(
    models=models,
    freq="H",
)

cross_validation_results = sf.cross_validation(
    df=df,
    step_size=24,
    h=24,
    n_windows=4,  
)

cross_validation.evaluate(
      losses = ["RMSE", "MAPE"]
)
```

or alternatively accept a list of functions?

``` python
from statsforecast.losses import MAPE, RMSE

cross_validation.evaluate(
      losses = [RMSE, MAPE]
)
```

The output would be some dataframe like object with the results of the selected loss functions like:

| model | metric | score |
|-------|--------|-------|
| ADIDA | RMSE   | 1     |
| ADIDA | MAPE   | 1.1   |
| ARCH  | RMSE   | 1.2   |
| ARCH  | MAPE   | 1.3   |

Obviously there may well be [insert good reason here], so if this contradicts goals, established conventions or priorities, let me know :)

**Comment by jmoralez** [comment]
Hey. It's because the metrics can be reused across packages, so it makes sense to have them on a separate package and use them in stats, ml and neural. We've actually migrated the losses to [utilsforecast](https://nixtla.github.io/utilsforecast/losses.html), which should feel more familiar, since they're implemented to take a dataframe with a structure like the output of the predict/cv steps. You can also look at the [evaluate function](https://nixtla.github.io/utilsforecast/evaluation.html) that computes several metrics at once.

Is `cross_validation` a dataframe in your example?

**Reply by DaveParr** [reply]
Thanks @jmoralez , makes perfect sense. My mistake for totally missing the existnce of `utilsforecast`!

**Reply by DaveParr** [reply]
@jmoralez is the intent to emphasise the use of this new `utilsforecast` package? 
If so, is there an interest in updating the example notebooks [like this one](https://nixtla.github.io/statsforecast/docs/models/autotheta.html#model-evaluation) to use this rather than `datasetsforecast`? 

Happy to have a pass at one or two of those if you think it would be worthwhile :)

**Reply by jmoralez** [reply]
Yes, we're migrating common things like plotting and losses to utilsforecast. The example [here](https://nixtla.github.io/statsforecast/docs/getting-started/getting_started_complete.html#evaluate-the-models-performance) already uses it, sadly in those model guides you're referring to that evaluation using datasetsforecast was copied over and over.

Feel free to contribute some migrations if you have time.
